src/Resource/LinkResource.php

```diff
@@ -52,6 +52,18 @@ public function getTargetPath()
     /**
      * {@inheritdoc}
      */
+    public function getTarget()
+    {
+        if (!$this->getRepository()) {
+            throw ResourceNotFoundException::forPath($this->getTargetPath());
```

Comment:
IMO this exception is misleading. The error is not that the target resource is not found, but that the link resource is not attached to a repository.

I think you should throw a RuntimeException here (and in similar cases).